lib/switchtec.c

```diff
@@ -93,6 +94,10 @@ struct switchtec_dev *switchtec_open(const char *device)
 
 	if (strchr(device, '/') || strchr(device, '\\')) {
 		ret = switchtec_open_by_path(device);
+		if (ret)
+			goto found;
+
+		ret = switchtec_open_uart(device);
```

Comment:
I'd very much prefer the way I told you to do it twice. Inside switchtec_open_by_path(), when you have already opened the file,  check isatty() and, if it is, then branch off into the uart code. This way we don't close the file and reopen or return an error when it's not an error.

Reply:
so for uart platform, it need call "switchtec_open_by_path", which is a function defined in Linux.c file?

Reply:
It already calls switchtec_open_by_path()... You need to change that function. (Which, yes, is in linux.c).

Reply:
change the function, pls comments.